app/utils/LocationFinder.scala

```diff
@@ -0,0 +1,98 @@
+package utils
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+import play.api.libs.ws._
+import play.api.Play.current
+
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration._
+import scala.math._
+import scala.util.{Success, Failure}
+
+object LocationFinder {
+  case class Location(lat: Double, lng: Double)
+  case class Place(name: String, location: Location)
+  case class Places(places: Seq[Place])
+
+
+  // Implicit Reads for case classes
+  implicit val locationReads : Reads[Location] = (
+    (JsPath \ "lat").read[Double] and
+    (JsPath \ "lng").read[Double]
+  )(Location.apply _)
+
+  implicit val placeReads : Reads[Place] = (
+    (JsPath \ "name").read[String] and
+    (JsPath \ "geometry" \ "location").read[Location]
+  )(Place.apply _)
+
+  // Explicit Reads for Places case class
+  val placesReads : Reads[Places] = (
+    (JsPath \ "results").read[Seq[Place]].map(Places(_))
+  )
+
+
+  // Concurrent execution context for Future
+  implicit val context = scala.concurrent.ExecutionContext.Implicits.global
+
+
+  // Find Euclidean distance between two GPS coordinates
```

Comment:
I think Euclidean distance is definitely good enough (we're not going to be worrying about distances > 10 miles maximum) but I'd thought about it greater-circle distance while I was doing this too. See #34.